src/test/java/uk/gov/companieshouse/chipsrestinterfacesconsumer/retry/MessageProducerImplUnitTest.java

```diff
@@ -48,22 +53,27 @@
     @InjectMocks
     private MessageProducerImpl messageProducerImpl;
 
+    @Captor
+    private ArgumentCaptor<Message> kafkaMessageCaptor;
+
     @Test
     void testSuccessfulWriteToTopic()
             throws ServiceException, ExecutionException, InterruptedException {
         when(producer.sendAndReturnFuture(any())).thenReturn(mockedFuture);
 
-        messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend());
-
+        messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend(), TEST_TOPIC);
         verify(mockedFuture, times(1)).get();
+        verify(producer, times(1)).sendAndReturnFuture(kafkaMessageCaptor.capture());
+        Message kafkaMessage = kafkaMessageCaptor.getValue();
+        assertEquals(TEST_TOPIC, kafkaMessage.getTopic());
     }
 
     @Test
     void testServiceExceptionIsThrownWhenSerializerThrowsIOException()
             throws IOException {
         doThrow(IOException.class).when(avroSerializer).serialize(any());
-
-        assertThrows(ServiceException.class, () -> messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend()));
+        assertThrows(ServiceException.class, () -> messageProducerImpl.writeToTopic(getDummyChipsRestInterfacesSend(), TEST_TOPIC));
+        verify(producer, times(0)).sendAndReturnFuture(kafkaMessageCaptor.capture());
```

Comment:
you have a capture here but you don't do anything with it. As you are verifying that it is not called, in this instance you can use any() instead of capture as you are not concerned with what value it is given.

Reply:
using any